src/ansi-c/c_typecheck_code.cpp

```diff
@@ -903,39 +922,91 @@ void c_typecheck_baset::typecheck_spec_assigns_condition(exprt &condition)
     throw 0;
 }
 
+void c_typecheck_baset::typecheck_conditional_targets(
+  exprt::operandst &targets,
+  const std::function<void(exprt &)> typecheck_target,
+  const std::string &clause_type)
+{
+  exprt::operandst tmp;
+  bool must_throw = false;
+
+  for(auto &target : targets)
+  {
+    if(!can_cast_expr<conditional_target_group_exprt>(target))
+    {
+      must_throw = true;
```

Comment:
I'm not sure this `must_throw` is even needed: I _believe_ (but would be happy to be proved wrong) that the type checker eventually fails when any output to `error` was produced.